src/components/App/App.js

```diff
@@ -2,25 +2,59 @@ import React, { useState, useEffect } from 'react';
 import { getTips, addTip } from '../../ApiCalls';
 import TipJar from '../TipJar/TipJar';
 import Form from '../Form/Form';
+import Error from '../Error/Error';
+import Loader from '../Loader/Loader';
 import './App.css';
 
 const App = () => {
   const [advice, setAdvice] = useState([]);
+  const [error, setError] = useState('');
+  const [loading, setLoading] = useState('Loading tips...');
+
+  // useEffect(() => {
+  //   getTips()
+  //   .then(data => setAdvice(data.rows))
+  // }, [])
 
   useEffect(() => {
-    getTips().then(data => setAdvice(data.rows))
-  }, [])
+    const fetchData = async () => {
+      setError('');
+ 
+      try {
+        const result = await getTips();
```

Comment:
Nice! glad you figured out the syntax for that....I had no idea how to do this with async / await rather than a bunch of .thens

Reply:
It actually works but today we learned that we shouldn't put an async in a useEffect...because it could cause an infinite loop. This is interesting, though, because this is an example I found that was recommended online and it works!